### PR TITLE
internal: fix race in test httpServer

### PR DESCRIPTION
**Pull request by jadekler**
Found this while running `TestHTTPToGRPCStatusMapping` 10,000 times. I just put a lock around the two usages that caused the test, and elided the other usages.

The multiple accesses happen because `TestHTTPToGRPCStatusMapping` iterates over a slice of test cases and calls setup/cleanUp during each test. setup happens inside a new goroutine.

I don't believe that these operations ever happen in parallel: cleanUp is deferred to the end of the test, whereas setup happens at the start of the test. I think the race detector just thinks it's a race. I think that's something that can happen with the detector? WDYT?

Here's the error this fixes:

```
=== RUN   TestHTTPToGRPCStatusMapping
==================
WARNING: DATA RACE
Read at 0x00c00043ce20 by goroutine 22:
  google.golang.org/grpc/internal/transport.(*httpServer).cleanUp()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2022 +0x3a
  google.golang.org/grpc/internal/transport.setUpHTTPStatusTest.func1()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2039 +0xe3
  google.golang.org/grpc/internal/transport.setUpHTTPStatusTest.func2()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2047 +0x6d
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:522 +0x3a
  testing.(*common).Fatalf()
      /usr/local/go/src/testing/testing.go:634 +0x94
  google.golang.org/grpc/internal/transport.setUpHTTPStatusTest()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2063 +0xabf
  google.golang.org/grpc/internal/transport.testHTTPToGRPCStatusMapping()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2080 +0x6a
  google.golang.org/grpc/internal/transport.TestHTTPToGRPCStatusMapping()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2075 +0xce
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Previous write at 0x00c00043ce20 by goroutine 7:
  google.golang.org/grpc/internal/transport.(*httpServer).start.func1()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:1983 +0xa0

Goroutine 22 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:878 +0x650
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1034 +0x2ee
  main.main()
      _testmain.go:160 +0x221

Goroutine 7 (finished) created at:
  google.golang.org/grpc/internal/transport.(*httpServer).start()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:1981 +0x6a
  google.golang.org/grpc/internal/transport.setUpHTTPStatusTest()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2058 +0x4f2
  google.golang.org/grpc/internal/transport.testHTTPToGRPCStatusMapping()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2080 +0x6a
  google.golang.org/grpc/internal/transport.TestHTTPToGRPCStatusMapping()
      /Users/deklerk/workspace/go/src/google.golang.org/grpc/internal/transport/transport_test.go:2075 +0xce
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
```

### CODE REVIEW

File: internal/transport/transport_test.go [hunk 1]
```diff
@@ -2052,6 +2058,7 @@ func setUpHTTPStatusTest(t *testing.T, httpStatus int, wh writeHeaders) (stream
 		t.Fatalf("Failed to listen. Err: %v", err)
 	}
 	server = &httpServer{
+		mu:         sync.Mutex{},
```
Comment (dfawley) [inline internal/transport/transport_test.go]: This is unnecessary.  It's automatically initialized to its zero value.

Comment (jadekler) [reply internal/transport/transport_test.go]: Done.

**Comment by jadekler** [comment]
Done

**Comment by jadekler** [comment]
Replacing with a mega PR.

**Pull request closed without merging**